### PR TITLE
Python access to Table Collection index

### LINKED ISSUE #4: TableCollection needs interface for indexes
We should provide access to the indexes in TableCollection, i.e., to the arrays themselves and to the ``build_indexes`` and ``drop_indexes`` methods.

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #4 

There is a separate issue for LWT (#870) so I'll do a new PR for that.

### CODE REVIEW

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/tskit-docs/916/)

**Review by jeromekelleher — APPROVED** [review]
LGTM

**Comment by benjeffery** [comment]
Waiting for https://github.com/tskit-dev/tskit/pull/921#discussion_r511880526 to be resolved before merging.

**Comment by benjeffery** [comment]
@jeromekelleher I've just realised (belatedly) that I need to do `set_index` for `from_dict` to apply the indexes from the dict from the lwt.

**Comment by jeromekelleher** [comment]
I think we should standardise on either calling this ``indexes`` or ``index``. I don't mind if there's a slight inconsistency in having ``has_index`` and ``build_index``. Arguably, this is the right thing because if we had multiple indexes, we might want to build only one of them (i.e., these functions would have an argument).

Since the file format has ``indexes`` I think it makes most sense to keep using this. 

Let's resolve this issue here before picking up #921

**Review by jeromekelleher — CHANGES_REQUESTED** [review]
I think it should be ``indexes`` through out, and let's add the functionality and tests for ``set_indexes`` here, to keep the lwt changes separate.

**Comment by benjeffery** [comment]
@jeromekelleher This now has write access to the indexes from the Python API and a new flag on `load_tables` to not rebuild indexes. Currently the default is `True` but that doesn't seem the right thing now that indexes are passed through the LWT. I'm looking into how much breakage there will be if the default is false.

**Comment by benjeffery** [comment]
I've changed the default to be not building indexes, the major change is that `TableCollection.tree_sequence` now builds the index if there is none.

File: python/tests/test_tables.py
Comment (jeromekelleher) [inline python/tests/test_tables.py:1316]: Should probably be ``TableCollectionIndexes`` for consistency

**Review by jeromekelleher — COMMENTED** [review]
Looks good, just a few minor inconsistencies

File: python/tests/test_tables.py
Comment (benjeffery) [reply python/tests/test_tables.py]: Fixed in https://github.com/tskit-dev/tskit/pull/916/commits/150dad1251e1329ce97824161e96a7e18c5ac60b

File: c/tskit/tables.h
Comment (benjeffery) [reply c/tskit/tables.h]: Fixed in https://github.com/tskit-dev/tskit/pull/916/commits/150dad1251e1329ce97824161e96a7e18c5ac60b

**Review by jeromekelleher — APPROVED** [review]

**Pull request merged**
